typescript/tuple_tree.ts

```diff
@@ -47,8 +47,8 @@ export class Reference<T, M> {
         }
     }
 
-    resolve(root: M): T {
-        return _getElementByPath<T>(this.reference, root)!;
+    resolve(root: M): T | undefined {
+        return _getElementByPath<T>(this.reference, root);
     }
```

Comment:
What's the use case here?

Reply:
Basically, in TS when traversing the model, you get to a reference object, let's call `ref`, if you want to get the pointed object you would do `ref.resolve(binary)`, however it's not guaranteed that the pointed paths exists, so to make sure the caller of `resolve` handles that case, we return `undefined` in case the path does not exist.